packages/trader/src/Modules/CFD/Containers/cfd-dashboard.jsx

```diff
@@ -39,139 +39,128 @@ const LoadTab = ({ children, is_loading, loading_component, ...props }) => {
-        },
-    };
-
-    async componentDidMount() {
-        this.updateActiveIndex(this.getIndexToSet());
-        this.openResetPassword();
-        this.props.onMount();
-    }
-
-    componentWillUnmount() {
-        this.props.onUnmount();
-    }
-
-    componentDidUpdate() {
-        this.updateActiveIndex();
-        this.props.checkShouldOpenAccount();
-
-        if (this.props.is_logged_in) {
+const CFDDashboard = props => {
+    const [is_demo_enabled, setIsDemoEnabled] = React.useState(false);
+    const [is_real_enabled, setIsRealEnabled] = React.useState(false);
+    const [active_index, setActiveIndex] = React.useState(0);
+    const [is_demo_tab, setIsDemoTab] = React.useState(true);
+    const [is_notification_loaded, setIsNotificationLoaded] = React.useState(false);
+    const [password_manager, setPasswordManager] = React.useState({
+        is_visible: false,
+        selected_login: '',
+        selected_account: '',
+        selected_account_type: '',
+        selected_account_group: '',
+        selected_server: '',
+    });
+
+    React.useEffect(() => {
+        updateActiveIndex(getIndexToSet());
+        openResetPassword();
+        props.onMount();
+        return () => {
+            props.onUnmount();
+        };
+    }, []);
+
+    React.useEffect(() => {
+        updateActiveIndex();
+        props.checkShouldOpenAccount();
+
+        if (props.is_logged_in) {
             ['demo', 'real'].forEach(account_type => {
                 const should_enable_tab =
-                    this.isSyntheticCardVisible(account_type) ||
-                    this.isFinancialCardVisible() ||
-                    this.isFinancialStpCardVisible();
+                    isSyntheticCardVisible(account_type) || isFinancialCardVisible() || isFinancialStpCardVisible();
 
-                const is_tab_enabled = this.state[`is_${account_type}_enabled`];
+                if (account_type === 'real' && is_real_enabled !== should_enable_tab) {
+                    setIsRealEnabled(should_enable_tab);
+                }
 
-                if (is_tab_enabled !== should_enable_tab) {
-                    this.setState({
-                        [`is_${account_type}_enabled`]: should_enable_tab,
-                    });
+                if (account_type === 'demo' && is_demo_enabled !== should_enable_tab) {
+                    setIsDemoEnabled(should_enable_tab);
                 }
             });
         }
-        const is_real_disabled = !this.state.is_real_enabled;
-        const is_demo_disabled = !this.state.is_demo_enabled;
-        if (!this.props.is_logged_in && (is_real_disabled || is_demo_disabled)) {
-            this.setState({
-                is_real_enabled: true,
-                is_demo_enabled: true,
-            });
+
+        if (!props.is_logged_in && (!is_real_enabled || !is_demo_enabled)) {
+            setIsRealEnabled(true);
+            setIsDemoEnabled(true);
         }
-    }
+    });
```

Comment:
don't we need to  add dependencies here for update?
ex: , [props.is_logged_in]

Reply:
Since the main code re-renders on every update of component (componentDidUpdate), one the reasons that I was getting the refreshing bug was having dependencies and I made the useEffect to run on every render of the component by removing the dependencies so that on every update based on the conditions, our component will re-render.  We will face the same issue if I put dependencies in our useEffect.

Reply:
I see, thanks for your explanation :pray: